src/wallet/transaction.h

```diff
@@ -305,6 +305,13 @@ class CWalletTx
     CWalletTx(CWalletTx const &) = delete;
     void operator=(CWalletTx const &x) = delete;
 };
+
+struct WalletTxOrderComparator {
```

Comment:
What do you think about a lambda function instead? Quite a bit less code, and imo about equally readable? As long as we're not using this comparator anywhere else the lambda would have my preference, but I'm okay with the current implementation too.

<details>
<summary> git diff </summary>

```diff
diff --git a/src/wallet/transaction.h b/src/wallet/transaction.h
index 27983e356..c36cbeb23 100644
--- a/src/wallet/transaction.h
+++ b/src/wallet/transaction.h
@@ -306,12 +306,6 @@ public:
     void operator=(CWalletTx const &x) = delete;
 };
 
-struct WalletTxOrderComparator {
-    bool operator()(const CWalletTx* a, const CWalletTx* b) const
-    {
-        return a->nOrderPos < b->nOrderPos;
-    }
-};
 } // namespace wallet
 
 #endif // BITCOIN_WALLET_TRANSACTION_H
diff --git a/src/wallet/wallet.cpp b/src/wallet/wallet.cpp
index 69957ed9e..ad20f332a 100644
--- a/src/wallet/wallet.cpp
+++ b/src/wallet/wallet.cpp
@@ -1906,8 +1906,9 @@ void CWallet::ResubmitWalletTransactions(bool relay, bool force)
         LOCK(cs_wallet);
 
         // First filter for the transactions we want to rebroadcast.
-        // We use a set with WalletTxOrderComparator so that rebroadcasting occurs in insertion order
-        std::set<CWalletTx*, WalletTxOrderComparator> to_submit;
+        // We use a set ordered by nOrderPos so that rebroadcasting occurs in insertion order
+        auto cmp = [](const CWalletTx* a, const CWalletTx* b) { return a->nOrderPos < b->nOrderPos; };
+        std::set<CWalletTx*, decltype(cmp)> to_submit(cmp);
         for (auto& [txid, wtx] : mapWallet) {
             // Only rebroadcast unconfirmed txs
             if (!wtx.isUnconfirmed()) continue;
```

</details>

Reply:
I think the comparator might actually be useful in other places too.